File: app/src/main/java/test/notificationreader/model/notifications/MessageNotification.java
package test.notificationreader.model.notifications;

import android.support.annotation.Nullable;

public class MessageNotification extends Notification {

    private String mName;
    private String mMessage;

    public MessageNotification(String text, @Nullable String aPackage) {
        super(text, aPackage);

        String[] splitedText = text.split(":");
        mName = splitedText[0].trim();
        mMessage = splitedText[1].trim();
    }

    public String getName() {
        return mName;
    }

    public String getMessage() {
        return mMessage;
    }
}

File: app/src/main/kotlin/test/notificationreader/model/notifications/MessageNotification.kt
package test.notificationreader.model.notifications

class MessageNotification(text: String, aPackage: String?) : Notification(text, aPackage) {
    val name: String
    val message: String

    init {
        val splitedText = text.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        name = splitedText[0].trim { it <= ' ' }
        message = splitedText[1].trim { it <= ' ' }
    }
}
